plugins/modules/gitlab_label.py

```diff
@@ -268,13 +268,18 @@ def update_label(self, var_obj):
             return True, True
         _label = self.gitlab_object.labels.get(var_obj.get('name'))
 
+        if (var_obj.get('new_name') is None) and (var_obj.get('color') is None):
+            self._module.fail_json(msg="Failed to update label, new_name or color must be present: %s " % _label)
```

Comment:
Why is this needed? This is a new requirement that wasn't there in the past (and thus a breaking change).

Reply:
Hi @felixfontein , looking at [Gitlab API docs](https://docs.gitlab.com/ee/api/labels.html) seems it was there also [in the past (this is from gitlab 15.6) ](https://archives.docs.gitlab.com/15.6/ee/api/labels.html#edit-an-existing-label) but the python library used by this module does not mention [this constraint](https://python-gitlab.readthedocs.io/en/stable/gl_objects/labels.html):

> Update a label for a project:
> 
> ```
> # change the name of the label:
> label.new_name = 'bar'
> label.save()
> # change its color:
> label.color = '#112233'
> label.save()
> ```


I've done a fast check with following code:
```python
_project = gl.projects.get(...)
_l = _project.labels.get('my-label')
_l.description = "new descript"
_l.save()
```
it works, without any "color" or "name", so I'm going to remove this check.

Reply:
My guess is that the documentation has been wrong for a longer time, and "At least one parameter is required, to update the label." is the only relevant part. I'd guess that the "yes if xxx is not provided" has been there before `description` and `priority` was added.

And in fact: https://gitlab.com/gitlab-org/gitlab-foss/-/commit/c8be7f1cf027d27bba50f6fa4fdeaee33e3f531f#fd69b4be4cca7d3b5760b77f0fc25ea1c57685ae_115_126

If you look at the code change in the same commit (further below), you can see that the original condition for the error message `'Required parameters "new_name" or "color" missing'` changed to also include `description`, without the error message being updated (line ~80 of lib/api/labels.rb). I guess this was later updated again by adding `priority`, without again updating the (already wrong) documentation of `name` and `color`.